godot_project/editor/controls/dockers/workspace_docker/e_workspace_settings_docker/controls/visibility_settings/visibility_settings.gd

```diff
@@ -79,8 +79,8 @@ func _on_show_hydrogens_toggle_toggled(button_pressed: bool) -> void:
 		return
 	
 	if new_h_visibility:
-		_workspace_context.enable_hydrogens_visualization()
+		_workspace_context.enable_hydrogens_visualization(false)
 	else:
-		_workspace_context.disable_hydrogens_visualization()
+		_workspace_context.disable_hydrogens_visualization(true)
```

Comment:
the definition of this function is `disable_hydrogens_visualization(in_clear_hydrogen_selection: bool)
So sane thing to do when hidding the hydrogens for me is to deselect them (since they cannot be seen by the user anyway). 

We don't want to mess with selection when enabling hydrogens rendering, there is no need for it